### PR TITLE
refactor(users): Add deeplinked tabs via <RouteTabs> and refactor /users/[userId] in support 

### PR DESCRIPTION
**Pull request by damassi**
This adds a new pattern for deep-linking into subpages within an app by doing a few things: 

- Takes advantage of Next's [dynamic path support](https://nextjs.org/docs/routing/dynamic-routes) which allows us to create directories like `pages/users/[userId]`, which will then match subpages within
- Adds a new `RouteTabs` component which can be used to display tab-based UI that are actually links
- Adds a layout pattern that we can copy over to other pages (see `UserLayout.tsx`) 
- Refactors the `/users/[userId]` app in support of the previous bullets

Look at the new file/folder structure closely; that's how we should be laying out pages/sub-apps going forward! 

routerlinks

### CODE REVIEW

File: src/pages/users/[userId]/components/UserLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Spacer, Text } from "@artsy/palette"
+import { UserNav } from "pages/users/[userId]/components/UserNav"
+import { graphql, useFragment } from "react-relay"
+import { UserLayout_user$key } from "__generated__/UserLayout_user.graphql"
+
+interface UserLayoutProps {
+  user: UserLayout_user$key
+}
+
+export const UserLayout: React.FC<UserLayoutProps> = ({ user, children }) => {
```
Comment (damassi) [inline src/pages/users/[userId]/components/UserLayout.tsx:10]: Simple layout

File: src/pages/users/[userId]/components/UserNav.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export const UserNav: React.FC = () => {
+  const router = useRouter()
+  const { userId } = router.query
+
+  return (
+    <RouteTabs>
+      <RouteTab name="Account" href={`/users/${userId}`}>
+        Account
+      </RouteTab>
```
Comment (damassi) [inline src/pages/users/[userId]/components/UserNav.tsx:12]: And this is how we can do deep-linkable tabs

File: src/pages/users/[userId]/components/UserDataTransfer.tsx [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+import { useUpdateUser } from "pages/users/[userId]/mutations/useUpdateUser"
+import { graphql, useFragment } from "react-relay"
+import { UserDataTransfer_user$key } from "__generated__/UserDataTransfer_user.graphql"
+
+interface UserDataTransferProps {
+  user: UserDataTransfer_user$key
+}
+
+export const UserDataTransfer: React.FC<UserDataTransferProps> = ({ user }) => {
+  const data = useFragment(
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/pages/users/[userId]/components/UserDataTransfer.tsx:12]: (Fixed a couple spots in this PR)

File: src/pages/users/[userId]/collector-profile.page.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { GetServerSideProps } from "next"
```
Comment (damassi) [inline src/pages/users/[userId]/collector-profile.page.tsx:1]: Note the full file path up above -- `[userId]/collector-profile.page.tsx`

**Review by artsyjian — APPROVED** [review]
nice!

File: src/pages/users/[userId]/components/UserDataTransfer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ovasdi) [reply src/pages/users/[userId]/components/UserDataTransfer.tsx:12]: >I know @mzikherman mentioned this recently, but we shouldn't ever pass unmasked data around to components. Even if a component only needed user.internalID, use a fragment. It's the only way to scale things over time without it becoming brittle.

@damassi Could you elaborate a bit and/or link to an example as far as what you mean by _unmasked data_? Would like to understand exactly what you are trying to convey here. Thanks!

[the remark this answers is not in the document]
Comment (ovasdi) [reply src/pages/users/[userId]/components/UserDataTransfer.tsx:12]: Oh. Do you mean like [this](https://github.com/artsy/forque/blob/3e24607a76d74f36d1ba1ca9686064ac67ea3d1b/src/pages/users/components/UserSecurity.tsx#L13)?

[the remark this answers is not in the document]
Comment (ovasdi) [reply src/pages/users/[userId]/components/UserDataTransfer.tsx:12]: And then accessing the value of ID [this way](https://github.com/artsy/forque/blob/3e24607a76d74f36d1ba1ca9686064ac67ea3d1b/src/pages/users/components/checkbox/UserDataTransfer.tsx#L34).

[the remark this answers is not in the document]
Comment (damassi) [reply src/pages/users/[userId]/components/UserDataTransfer.tsx:12]: Sure thing:

```tsx
const App = () => {
  const userData = useFragment(graphql`
    fragment App_user on User {
      name 
    }
  `
  
  return (
    <UserName user={userData} />
  )
}

const UserName = ({ user }) => {
  return <div>{user.name}</div>
}
```

In the above, `UserName` should be requesting its own data via `useFragment`, not receiving it by having it passed as props. All data dependencies for all components should be co-located.

[the remark this answers is not in the document]
Comment (ovasdi) [reply src/pages/users/[userId]/components/UserDataTransfer.tsx:12]: Thank you!

**Pull request merged**